view/adminhtml/web/js/validation.js

```diff
@@ -0,0 +1,22 @@
+require([
+        'jquery',
+        'mage/translate',
+        'jquery/validate'],
+    function($){
+        $.validator.addMethod(
+            "reach-default-hs-code-validator", function (hscode) {
+                //Got help/hints from
+                //https://magento.stackexchange.com/questions/95171/magento-2-form-validation
+                //https://magento.stackexchange.com/questions/139023/custom-field-validation-in-system-xml
+                //https://regex101.com/
+                //This routine is enforcing good enough check on length so we possibly can remove the built in
+                // validation of the form `validate-length minimum-length-2 maximum-length-20` from system.xml.
+                //Does accept two consecutive special characters. Also allows mix and match of `.` and `-` in the
+                //hscode string - made it that way after talking (on Slack) to the team.
+                //Check is done only at frontend level
+                //Additionally the check could be done/enforced when getting saved in db (in a path way bypassing UI)
+                hscode = hscode.trimRight();
```

Comment:
do we care about any whitespace at the beginning of the hscode? It would be silly but a user could do "  12.123.B12  "

Reply:
Yeah I was wondering should I take the requirement statement of it starts with two number way too literally or not.  With that hesitation in mind I took it too literally and trimmed space only from the right hand side. It is super easy to trim from both ends and I am more than happy to do so Lizzy. Thanks!

Reply:
yay!